### PR TITLE
Allow `{{app}}` templating in producer messages

### PR DESCRIPTION
**Pull request by ypaq**

### CODE REVIEW

**Comment by ypaq** [comment]
cc @cyx

**Comment by cyx** [comment]
LGTM

**Comment by cyx** [comment]
@dmcinnes @gregburek ^

**Comment by gregburek** [comment]
This looks pretty cool @heroku/dod-core and @heroku/dod-kcz for visibility on sending notifications to customers via telex, we can use `{{app}}` for telex to filling.

**Comment by cyx** [comment]
For completeness, I indicated to @ypaq that we should make sure the title/body actually has the `{{app}}` substitution before making the API call.

**Comment by spofdamon** [comment]
:+1:

File: lib/mediators/messages/creator.rb [hunk 1]
```diff
@@ -17,7 +17,7 @@ def initialize(producer:, title:, body:, action_label:, action_url:, target_type
 
     def call
       Message.new(@args).tap do |msg|
-        if msg.target_type == Message::APP
+        if msg.target_type == Message::APP && (msg.title.include?(APP) || msg.body.include?(APP))
```
Comment (cyx) [inline lib/mediators/messages/creator.rb]: Perfecto

**Pull request merged**